model/encodable/keys.go

```diff
@@ -173,6 +182,26 @@ func (pub *RandomBeaconPubKey) UnmarshalJSON(b []byte) error {
 	return err
 }
 
+func (pub RandomBeaconPubKey) MarshalCBOR() ([]byte, error) {
+	if pub.PublicKey == nil {
+		return cbor.Marshal(nil)
+	}
+	return cbor.Marshal(toHex(pub.Encode()))
+}
+
+func (pub *RandomBeaconPubKey) UnmarshalCBOR(b []byte) error {
```

Comment:
is there a reason why `MarshalXXX` has a non-pointer receiver while `UnarmshalXXX` has a pointer receiver?

Reply:
@vishalchangrani thanks for pointing this out!

I must admit that I just copy and pasted from the `MarshalMsgpack()` functions and obviously did not pay enough attention :-) However, I spent several hours this morning trying to make the receivers consistently pointers but without luck.

It seems like the `MarshalXXX()` functions need NOT to have a receiver pointer otherwise the marshalling underlying marshalling gets messed up. I tried to 'unmess' it etc but without luck :-(

You can read about what I tried here [1] in my notes.

I also looked at all the receivers in the code base -- nothing to do with this PR -- to see how many are 'uneven' so to speak. There are quite a few and maybe there should be an issue created to somehow audit them? Maybe it would be good to have some kind of mechanism to 'mark' (with a source comment?) those uneven receivers which are uneven for a reason? We could then create automation to auto detect the unmarked ones?

Through examining the code I also noticed some `CBOR` related production and tests for the flow service events which I didn't created and I just added that.

[1] https://www.notion.so/dapperlabs/Continued-2-e4519b82dfb54fc8b78ca9374eeaf49c

Reply:
I would say keep it mixed then for this PR. Thanks for looking into it.